base/dates/parse.jl

```diff
@@ -115,3 +115,70 @@ function rfixwidth(num, n)
     length(s) < n ? lpad(s, n, 0) : s[end - n + 1:end]
 end
 
+function Base.parse(::Type{DateTime}, s::AbstractString, df::DateFormat{Symbol("yyyy-mm-dd\\THH:MM:SS.s")})
```

Comment:
This is what I was trying to avoid by having the number of times a token is repeated and fixwidth in the type parameters - you could just dispatch on the format object's `typeof`. This is fine too, and probably better.